pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/MultiStageOperator.java

```diff
@@ -220,6 +229,25 @@ protected Map<String, Plan.ExplainNode.AttributeValue> getExplainAttributes() {
     return Collections.emptyMap();
   }
 
+  private long getGcTimeMillis() {
```

Comment:
Could the logic in this method be centralized into `ThreadResourceSnapshot` / `ThreadResourceUsageProvider`. (or any other of those relevant classes) instead?